### PR TITLE
Reflective vest needs to be equipped to work

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

Added a SlotFlags field on ReflectComponent so that anything with the component, if the SlotFlag is set to anything besides None, has to be equipped in that indicated slot to work.

Default is set to none, so things set to none don't go through the check.
Things set to anything besides none, get checked.

## Why / Balance

People were holding their reflective vests in their hand and using it as an infinite health shield

## Technical details

I already explained it!

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/user-attachments/assets/3a5040e6-652b-4d8c-8f13-053ee9c4bfdd

## Breaking changes

not breaking

**Changelog**

:cl:
- tweak: Reflective vests now need to be equipped to receive its benefits.

### CODE REVIEW

File: Content.Shared/Weapons/Reflect/ReflectComponent.cs [hunk 1]
```diff
@@ -8,25 +9,32 @@ namespace Content.Shared.Weapons.Reflect;
 /// Uses <c>ItemToggleComponent</c> to control reflection.
 /// </summary>
 [RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
-public sealed partial class ReflectComponent : Component
+public sealed partial class ReflectComponent : Component, IClothingSlots
```
Comment (EmoGarbage404) [inline Content.Shared/Weapons/Reflect/ReflectComponent.cs:12]: i would say just remove the interface and give the variable a more descriptive name.

File: Content.Shared/Weapons/Reflect/ReflectComponent.cs [hunk 2]
```diff
@@ -8,25 +9,32 @@ namespace Content.Shared.Weapons.Reflect;
...
-    [DataField("soundOnReflect")]
+    [DataField]
     public SoundSpecifier? SoundOnReflect = new SoundPathSpecifier("/Audio/Weapons/Guns/Hits/laser_sear_wall.ogg");
+
+    /// <summary>
+    /// The reflect component only works if the item is equipped
+    /// in this inventory slot
+    /// </summary>
+    [DataField]
+    public SlotFlags Slots { get; set; } = SlotFlags.NONE;
```
Comment (EmoGarbage404) [inline Content.Shared/Weapons/Reflect/ReflectComponent.cs:37]: using NONE for the default is kinda strange and i think itd be more logical if it was nullable and then null was the value for "this isn't tied to a slot"

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Content.Shared/Weapons/Reflect/ReflectSystem.cs [hunk 3]
```diff
@@ -171,6 +178,16 @@ private void OnReflectHitscan(EntityUid uid, ReflectComponent component, ref Hit
             return false;
         }
 
+        // if the item has a designated inventory slot it has to be equipped into, for it to reflect
+        // check the user's inventory to see if the item is in the proper slot
+        if (reflect.Slots != SlotFlags.NONE &&
+            TryComp<InventoryComponent>(user, out var inventoryComp) &&
+            !_inventorySystem.TryGetInventoryEntity<ReflectComponent>((user, inventoryComp), out _))
```
[the remark this answers is not in the document]
Comment (Plykiya) [reply Content.Shared/Weapons/Reflect/ReflectSystem.cs:185]: TryGetInventoryEntity only retrieves entities with the component that are in their proper slot but only if the IClothingSlot interface is implemented, which is why I can't get rid of it or make it nullable guahh

**Comment by metalgearsloth** [comment]
@Plykiya is this still getting worked on

**Comment by Plykiya** [comment]
> @Plykiya is this still getting worked on

depends, if I have to make it not use the interface then I probably won't continue it, if I just have to migrate the duplicate code into a function I can do that

**Pull request closed without merging**